src/msix/AppxManifestObject.cpp

```diff
@@ -368,6 +368,33 @@ namespace MSIX {
         return static_cast<HRESULT>(Error::NotImplemented);
     }
 
+    // IAppxManifestReader4
+    HRESULT STDMETHODCALLTYPE AppxManifestObject::GetOptionalPackageInfo(IAppxManifestOptionalPackageInfo **optionalPackageInfo) noexcept try
+    {
+        ThrowErrorIf(Error::InvalidParameter, (optionalPackageInfo == nullptr || *optionalPackageInfo != nullptr), "bad pointer.");
+
+        struct _context
+        {
+            std::string mainPackageName;
+        };
+        _context context = {};
+
+        // Parse MainPackageDependency elements
+        XmlVisitor visitorMainPackageDependencies(static_cast<void*>(&context), [](void* c, const ComPtr<IXmlElement>& dependencyNode)->bool
+        {
+            _context* context = reinterpret_cast<_context*>(c);
+            if (context->mainPackageName.empty())

```

Comment:
>if (context->mainPackageName.empty()) [](start = 12, length = 37)

Instead of checking this, you can just return false when you find the first one, ending the for each.